.github/workflows/dockerhub.yml

```diff
@@ -8,9 +8,27 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: trigger
-      run: curl -X POST https://hub.docker.com/api/build/v1/source/1813a660-2ee5-4583-a238-dd54e9a6ebac/trigger/c8cd9f1f-f269-45bc-9750-a08327257f62/call/
-    
-    
-
-
+      - name: checkout code
+        uses: actions/checkout@v2
+      - name: install buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v1
+        with:
+          version: latest
+      - name: login to docker hub
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+      - name: build and push the image (master branch)
+        run: |
+          docker buildx build \
+            --push \
+            --tag neilpang/acme.sh:latest \
+            --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 .
+        if: ${{ github.ref == 'refs/heads/master' }}
+      - name: build and push the image (dev branch)
+        run: |
+          docker buildx build \
+            --push \
+            --tag neilpang/acme.sh:dev \

```

Comment:
please do not hardcode `dev` here.
I want to make it work for any branches other than "master".
I also need it work for a "Tag" for a "Release".

Reply:
in that case, would it make sense to just tag "latest" for master branch, and for other branches to tag with the branch name? Maybe even double tagging and adding the git sha?

Reply:
Yes,   Only the master is tagged as "latest".
Any other branches will be tagged as branch name.
And for any release,  it will be tagged as release number, like "neilpang/acme.sh:2.8.7"

Reply:
If it's easier, I'm ok to split the workflow files in to multiple ones.
One for master, one for branches, one for releases, etc.   As long as it's easier for you.

Reply:
I think it should be fine doing it on a single workflow, but I'm gonna need to try this on my fork first so I can test it out and then update this PR! I hope to have the changes done in the next hour or so!

Reply:
Cool, take your time.   I hope it won't become complicated in just one single file.